pandas/_libs/tslibs/timezones.pyx

```diff
@@ -210,6 +217,8 @@ cdef inline bint is_fixed_offset(tzinfo tz):
             return 1
         else:
             return 0
+    elif is_zoneinfo(tz):
+        return False
```

Comment:
Why `False` when the other return values are all `0` or `1`?

Also, this isn't quite right. You can tell if a `ZoneInfo` has a fixed offset by passing `None` to `utcoffset`; if it returns a value, that's the fixed offset.

This also works for `pytz`, `dateutil.tz.tzoffset` and `datetime.timezone`. It won't work for `dateutil.tz.gettz("UTC")` right now, but once [the `zoneinfo` backport](https://github.com/dateutil/dateutil/pull/1130) is merged, it will.

This whole function can probably be simplified to: `return bool(tz.utcoffset(None) is not None)`, or (to handle the fixed offset `tzfile` cases for `dateutil <= 2.8.2`):

```python
if treat_tz_as_dateutil(tz):
    if len(tz._trans_idx) == 0 and len(tz._trans_list) == 0:
        return 1
    else:
        return 0

return 0 if tz.utcoffset(None) is None else 1
```

Of course, even that will break when the next version of `dateutil` comes out, so you probably want some version pinning in place (like for `python-dateutil < 3.0`).

Reply:
> Why False when the other return values are all 0 or 1?

No good reason; I'll change it to match.

> This whole function can probably be simplified to [...]

Thanks.  I'll probably make a dedicated branch to both Do This Right and audit our usages of is_fixed_offset, which I don't think we're very consistent about.